com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/validations/CobolLanguageEngineFacade.java

```diff
@@ -68,23 +75,60 @@ public AnalysisResult analyze(
     return toAnalysisResult(engine.run(uri, text, textDocumentSyncType.toString()), uri);
   }
 
+  /**
+   * Don't analyze the document if it is empty or contains only sequence area
+   *
+   * @param text - document to analyze
+   * @return true if the document is empty from the engine point of view
+   */
   private static boolean isEmpty(String text) {
     return text.length() <= FIRST_LINE_SEQ_AND_EXTRA_OP;
   }
 
-  private static List<Diagnostic> convertErrors(List<SyntaxError> errors, String uri) {
-    return errors.stream()
-        .filter(errorOnlyFromCurrentDocument(uri))
-        .map(toDiagnostic())
-        .collect(toList());
+  private AnalysisResult toAnalysisResult(ResultWithErrors<SemanticContext> result, String uri) {
+    SemanticContext context = result.getResult();
```

Comment:
Is context non-null all the time ?

Reply:
Yes, it is either fulfilled or empty. Added annotations to assure that.